### ISSUE TITLE
the poor performance when i adjust the args.out_size=512

### ISSUE THREAD

**Issue by ShChen233** [mazurowski-lab/finetune-SAM#4]
![image](https://github.com/mazurowski-lab/finetune-SAM/assets/143151549/1a7356c8-01a8-4da8-8063-ed8d2e6665ee)
![image](https://github.com/mazurowski-lab/finetune-SAM/assets/143151549/ef1020fd-ffe7-4a30-bfa0-c9487349c3fd)
Previously, I used the code from the Medical-SAM-Adapter project (https://github.com/KidsWithTokens/Medical-SAM-Adapter) to perform multi-organ segmentation on the Synapse dataset, achieving an average Dice accuracy of 89%. However, after adjusting related parameters in the finetune-SAM project (such as setting out_size=512) and training using the adapter method, I found a significant decrease in the Dice score.
Can you provide some advice? Thank you very much.

**Comment by Guhanxue** [comment]
aha, May i ask if you adjust your output channel of the network? 

Also, It might bc current version of dataset/dataloader have not support multi-class segmentation version.  

I was using two different dataset.py before but i thought it was not very organized and elegant. Thus i am planning to combine all my dataset related codes into one Class. i am still organize that part of code (i am so sorry for the inconvience). Do you mind giving me around 1 day to update my repo, i will send you an msg when it is fixed.

**Comment by ShChen233** [comment]
Thank you very much for your detailed explanation. I have made an adjustment to the output channel of the network by setting the num_classes=args.n_classes.
Furthermore, I have recently embarked on a research project involving the adaptation of SAM to multi-organ segmentation. If it's convenient for you, could we possibly exchange contact information, such as WeChat or Email?hhh.  Perhaps we can collaborate in the future~

**Comment by ShChen233** [comment]
![image](https://github.com/mazurowski-lab/finetune-SAM/assets/143151549/edbdd524-c9a2-48ce-8d57-9b24443e35b8)

**Comment by Guhanxue** [comment]
cooooool, what is your wechat?

**Comment by ShChen233** [comment]
emmm，It may not be convenient to show wechat id under the Issues? hhh. If it's convenient for you, could you provide your email address? My email is shchen@zju.edu.cn

**Comment by Guhanxue** [comment]
Hi, i updated the support for multi-cls segmentation.  Please free feel to test it.

**Comment by ShChen233** [comment]
I'm sorry, I forgot that I had updated my email address... It should be 22332040@zju.edu.cn.  @Guhanxue 。And thank you very much for your detailed explanation again~
